event/p2p.go

```diff
@@ -27,4 +27,22 @@ var (
 	// from a Qri peer
 	// payload will be a p2p.Message
 	ETP2PMessageReceived = Type("p2p:MessageReceived")
+	// ETP2PProfileRequestReceived fires when the profile exchange service
+	// recieves a request to begin a profile exchange
+	ETP2PProfileRequestReceived = Type("p2p:Profile exchange request  received")
+	// ETP2PProfileRequestSent fires when the node sends a request to another
+	// instance to begin a profiles exchange
+	ETP2PProfileRequestSent = Type("p2p:Profile exchange request sent")
+	// ETP2PProfileSendSuccess fires when the profile exchange service
+	// successfully completes a profile exchange
+	ETP2PProfileSendSuccess = Type("p2p:Profile sent successfully")
+	// ETP2PProfileSendFailed fires when the profile exchange service
+	// failes tp completes a profile exchange
+	ETP2PProfileSendFailed = Type("p2p:Profile send failed")
+	// ETP2PProfileReceiveSuccess fires when the profile exchange service
+	// successfully completes a profile exchange
+	ETP2PProfileReceiveSuccess = Type("p2p:Profile received successfully")
+	// ETP2PProfileReceiveFailed fires when the profile exchange service
+	// failes tp completes a profile exchange
+	ETP2PProfileReceiveFailed = Type("p2p:Profile receive failed")
```

Comment:
These events are too fine-grained in my opinion. My rule when defining an event: come up with a concrete use case that can't make use of an existing event, and isn't a test. To me these events are exposing too much of the internals of the profile exchange service.

We should aim to make events as general as possible, and err on the side of _not_ defining them because they're easy to add, hard to remove, and a potential source of slowness.

How about this instead:

```go
const (
  // ...

  // ETP2PQriPeerConnected fires whenever a peer-2-peer connection that
  // supports the qri protocol is established
  // payload is a profile.Profile
  // subscribers cannot block the publisher
  ETP2PQriPeerConnected = Type("p2p:QriPeerConnected")
  // ETP2PQriPeerDisconnected fires whenever a qri peer-2-peer connection
  // is closed
  // payload is a profile.Profile
  // subscribers cannot block the publisher
  ETP2PQriPeerDisconnected = Type("p2p:QriPeerConnected")
)
```

We can emit `ETP2PQriPeerConnected` in what is presently `ETP2PProfileReceiveSuccess`, and wire up a new `libp2p.Event` handler for disconnects that calls a method on `*ProfileService` that checks if the disconnected peer is a qri peer and emits the event

Reply:
Yes, I hear you. Definitely using them at crutches for the test-plan test, wanting as much incite into what is going on with the profile exchange as possible to debug. Will alter.